ext/oci8/oci8_interface.c

```diff
@@ -608,30 +570,23 @@ PHP_FUNCTION(oci_lob_size)
    Writes data to current position of a LOB */
 PHP_FUNCTION(oci_lob_write)
 {
-	zval *tmp, *z_descriptor = getThis();
+	zval *tmp, *z_descriptor;
 	php_oci_descriptor *descriptor;
 	size_t data_len;
 	zend_long write_len = 0;
 	ub4 bytes_written;
 	char *data;
 
-	if (getThis()) {
-		if (zend_parse_parameters(ZEND_NUM_ARGS(), "s|l", &data, &data_len, &write_len) == FAILURE) {
-			RETURN_THROWS();
-		}
+	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "Os|l", &z_descriptor, oci_lob_class_entry_ptr, &data, &data_len, &write_len) == FAILURE) {
+		RETURN_THROWS();
+	}
 
-		if (ZEND_NUM_ARGS() == 2) {
-			data_len = MIN((zend_long) data_len, write_len);
-		}
+	if (getThis() && ZEND_NUM_ARGS() == 2) {
```

Comment:
These consecutive checks everywhere could be optimized a tiny little bit by either:
- using `else if` clauses
- extracting `getThis()` in an outer condition so that this function is not called multiple times

However, all the `ZEND_NUM_ARGS() == ...` checks indicate that the default value handling is not right - which is something we want to fix later. That's why I'm also ok with leaving these conditions as you have them currently.